# Проект из мастерской

В данной части работы проводится подготовка данных к созданию дашборда

<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Импорт-данных" data-toc-modified-id="Импорт-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Импорт данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span></li><li><span><a href="#Работа-со-столбцом--job_description" data-toc-modified-id="Работа-со-столбцом--job_description-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Работа со столбцом  job_description</a></span></li><li><span><a href="#Код-для-сопоставления-и-проверки-данных" data-toc-modified-id="Код-для-сопоставления-и-проверки-данных-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Код для сопоставления и проверки данных</a></span></li><li><span><a href="#Дашборд" data-toc-modified-id="Дашборд-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Дашборд</a></span></li></ul></div>

## Импорт данных 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from plotly import graph_objects as go 
import plotly.express as px
from statsmodels. stats.weightstats import ztest as ztest
from scipy import stats as st
import numpy as np
import math as mth
import warnings

warnings.filterwarnings('ignore')

In [2]:
#data=pd.read_csv('F:\!Браузер загрузка\linkedin (3).csv')

In [3]:
data = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vSsYtLzJcZo1TVYUSrTc8LwxEt0O1BgXM1B8vCmlME1Fn-c9lxIrtzHuLjmGZ1sUXOz2Ev_T4fPeaCm/pub?gid=1073518690&single=true&output=csv')

In [4]:
data.head(15)

title       location         country  \
0                      Data Analyst          Basel     Switzerland   
1          Data Analyst - Logistics       Coventry  United Kingdom   
2          Data Analyst - Logistics       Coventry  United Kingdom   
3   Data Analyst (Space & Planning)   South Molton  United Kingdom   
4                      Data Analyst         Lugano     Switzerland   
5          Data Analyst - Logistics    Southampton  United Kingdom   
6                      Data Analyst          Leeds  United Kingdom   
7                      Data Analyst       Nuneaton  United Kingdom   
8                      Data Analyst          Paris          France   
9     Data Analyst - Hybrid Working      Cambridge  United Kingdom   
10                     Data Analyst  West Midlands  United Kingdom   
11          Commercial Data Analyst       Schiphol     Netherlands   
12               Asset Data Analyst        Chester  United Kingdom   
13            Data Analyst - Hybrid      Cambridge  United Kingdom   
14                     Data Analyst    Craven Arms  United Kingdom   

   employment_type                                       company_name  \
0          On-site           PharmiWeb.Jobs: Global Life Science Jobs   
1          On-site                               Resolute Recruitment   
2          On-site                               Resolute Recruitment   
3          On-site                                Mole Valley Farmers   
4          On-site                                            FORFIRM   
5          On-site                               Butler, Bridge & May   
6          On-site                     Maria Mallaband Care Group Ltd   
7           Hybrid                                        Kelly Group   
8          On-site                                              eXalt   
9          On-site                                         Blue Arrow   
10         On-site                                     Mexa Solutions   
11         On-site                               Royal Schiphol Group   
12         On-site  networx | Recruitment Software & Services by IRIS   
13         On-site                                         Blue Arrow   
14   not specified  networx | Recruitment Software & Services by IRIS   

             employee_qty                  company_field  \
0         11-50 employees        Staffing and Recruiting   
1           not specified                  not specified   
2           not specified                  not specified   
3           not specified                  not specified   
4           not specified                  not specified   
5           not specified                  not specified   
6           not specified                  not specified   
7           not specified                  not specified   
8     501-1,000 employees  IT Services and IT Consulting   
9           not specified                  not specified   
10          not specified                  not specified   
11  1,001-5,000 employees          Airlines and Aviation   
12          not specified                  not specified   
13          not specified                  not specified   
14          not specified                  not specified   

                                               skills  \
0                                                None   
1                                                None   
2                                                None   
3                                                None   
4                                                None   
5                                                None   
6                                                None   
7                                                None   
8   <span class="visually-hidden"><!-- -->Skills: ...   
9                                                None   
10                                               None   
11  <span class="visually-hidden"><!-- -->Skills: ...   
12                                    

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 998 entries, 0 to 997
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title            998 non-null    object 
 1   location         998 non-null    object 
 2   country          998 non-null    object 
 3   employment_type  998 non-null    object 
 4   company_name     996 non-null    object 
 5   employee_qty     998 non-null    object 
 6   company_field    998 non-null    object 
 7   skills           998 non-null    object 
 8   job_description  997 non-null    object 
 9   applicants       838 non-null    float64
dtypes: float64(1), object(9)
memory usage: 78.1+ KB


Есть пропуски в столбце с кандидатами, тип данных везде верный

##  Предобработка данных

Оценим список уникальных значений в столбце с типом занятости

In [6]:
data['employment_type'].unique()

array(['On-site', 'Hybrid', 'not specified', 'Remote'], dtype=object)

Их всего 4: в офисе, гибрид, не указано, удаленная

Оценим столбец с количеством работников

In [7]:
data['employee_qty'].unique()

array(['11-50 employees', 'not specified', '501-1,000 employees',
       '1,001-5,000 employees', '51-200 employees', '10,001+ employees',
       '1-10 employees', '201-500 employees', '5,001-10,000 employees',
       'Retail Apparel and Fashion',
       'See how you compare to 9 applicants. Try Premium for free',
       'Svein Grande is hiring for this job',
       'See how you compare to 19 applicants. Try Premium for free',
       'See how you compare to 22 applicants. Try Premium for free',
       'Romain GUIHENEUF is hiring for this job',
       'See how you compare to 13 applicants. Try Premium for free',
       'See recent hiring trends for Devonshire Hayes Recruitment Specialists Ltd. Try Premium for free',
       'See how you compare to 10 applicants. Try Premium for free',
       'See how you compare to 4 applicants. Try Premium for free'],
      dtype=object)

In [8]:
s=['11-50 employees', 'not specified', '501-1,000 employees',
       '1,001-5,000 employees', '51-200 employees', '10,001+ employees',
       '1-10 employees', '201-500 employees', '5,001-10,000 employees']
display(s)

['11-50 employees',
 'not specified',
 '501-1,000 employees',
 '1,001-5,000 employees',
 '51-200 employees',
 '10,001+ employees',
 '1-10 employees',
 '201-500 employees',
 '5,001-10,000 employees']

In [9]:
#Ищем строки, значения которых нет в списке s, и заменяем значение
data.loc[~data['employee_qty'].isin(s), 'employee_qty'] = 'not specified'

In [10]:
data.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
1         Data Analyst - Logistics      Coventry  United Kingdom   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description  applicants  
0                              What You Will Achi...        47.0  
1                                                NaN         NaN  
2                               Data Analyst - Lo...         NaN  
3                                Salary:     To b...         NaN  
4                                FORFIRM     is p...         NaN

проверка:

In [11]:
data['employee_qty'].unique()

array(['11-50 employees', 'not specified', '501-1,000 employees',
       '1,001-5,000 employees', '51-200 employees', '10,001+ employees',
       '1-10 employees', '201-500 employees', '5,001-10,000 employees'],
      dtype=object)

In [12]:
data['company_field'].unique()

array(['Staffing and Recruiting', 'not specified',
       'IT Services and IT Consulting', 'Airlines and Aviation',
       'Medical Equipment Manufacturing',
       'Information Technology & Services',
       'Transportation, Logistics, Supply Chain and Storage',
       'Real Estate', 'Financial Services', 'Entertainment Providers',
       'Software Development', '51-200 employees',
       'Technology, Information and Internet', 'Manufacturing',
       'Human Resources', 'Banking', 'Advertising Services',
       'Public Relations and Communications Services',
       'Packaging and Containers Manufacturing', '11-50 employees',
       'Telecommunications', 'Newspaper Publishing', 'Higher Education',
       'Internet Marketplace Platforms', 'Retail', 'Insurance',
       'Leisure, Travel & Tourism', 'Human Resources Services',
       'Motor Vehicle Manufacturing',
       'Personal Care Product Manufacturing', 'Utilities',
       'International Trade and Development', 'Information Services'

In [13]:
data.loc[data['company_field'].str.contains('Try Premium for free'), 'company_field'] = 'not specified'

In [14]:
#Ищем строки, значения которых нет в списке s, и заменяем значение
data.loc[data['company_field'].isin(s), 'company_field'] = 'not specified'

In [15]:
data['company_field']=data['company_field'].str.replace('Romain GUIHENEUF is hiring for this job','not specified')

In [16]:
data['company_field'].unique()

array(['Staffing and Recruiting', 'not specified',
       'IT Services and IT Consulting', 'Airlines and Aviation',
       'Medical Equipment Manufacturing',
       'Information Technology & Services',
       'Transportation, Logistics, Supply Chain and Storage',
       'Real Estate', 'Financial Services', 'Entertainment Providers',
       'Software Development', 'Technology, Information and Internet',
       'Manufacturing', 'Human Resources', 'Banking',
       'Advertising Services',
       'Public Relations and Communications Services',
       'Packaging and Containers Manufacturing', 'Telecommunications',
       'Newspaper Publishing', 'Higher Education',
       'Internet Marketplace Platforms', 'Retail', 'Insurance',
       'Leisure, Travel & Tourism', 'Human Resources Services',
       'Motor Vehicle Manufacturing',
       'Personal Care Product Manufacturing', 'Utilities',
       'International Trade and Development', 'Information Services',
       'Computer Games', 'Retail Heal

In [17]:
data['applicants']=data['applicants'].fillna('not specified')

Проверим данные на наличие дубликатов

In [18]:
data.duplicated().sum()

112

In [19]:
data.shape[0]

998

Всего 112 дубликатов, удалим их

In [20]:
data=data.drop_duplicates()

Проверим отдельные значения

In [21]:
data[data['company_name']=='Resolute Recruitment']

title  location         country employment_type  \
1  Data Analyst - Logistics  Coventry  United Kingdom         On-site   
2  Data Analyst - Logistics  Coventry  United Kingdom         On-site   

           company_name   employee_qty  company_field skills  \
1  Resolute Recruitment  not specified  not specified   None   
2  Resolute Recruitment  not specified  not specified   None   

                                     job_description     applicants  
1                                                NaN  not specified  
2                               Data Analyst - Lo...  not specified

In [22]:
ks=data[data['company_name']=='Resolute Recruitment']

In [23]:
ks['job_description'].unique()

array([nan,
       '                             Data Analyst - Logistics ~ Permanent Role ~ Mon-Fri ~ Immediate Start ~Locations: Coventry, CV3Salary: 35,000 - 42,000 Per Year Shift Pattern: Monday to Friday 08:00 - 16:00*The correct candidate will also be able to work night shifts when required to observe this side of the business. Once the initial training period is complete you will be able to work from home as and when required. Job role: Analyse business data and provide detailed reports and reccomendations. Provide the business with regular reports to Quality Performance Supply key information to senior reviewers to assist decision making Ensuring actions are taken to speed up the issue resolution process Analyse general business processes to target cost saving &amp; time saving efficiencies.. Report to business management and reccomend any changes. Monday to Friday Days, however the correct candidate must be prepared to work night shifts to look at this area of the business on 

удалим

In [24]:
data = data.drop(index=[1])
data.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   
5         Data Analyst - Logistics   Southampton  United Kingdom   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   
5         On-site                      Butler, Bridge & May    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   
5            not specified   None   

                                     job_description     applicants  
0                              What You Will Achi...           47.0  
2                               Data Analyst - Lo...  not specified  
3                                Salary:     To b...  not specified  
4                                FORFIRM     is p...  not specified  
5                               Location: Southam...  not specified

Также у нас есть 2 проупска в названии компаний, удалим

In [25]:
data=data.dropna()

In [26]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 883 entries, 0 to 997
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   title            883 non-null    object
 1   location         883 non-null    object
 2   country          883 non-null    object
 3   employment_type  883 non-null    object
 4   company_name     883 non-null    object
 5   employee_qty     883 non-null    object
 6   company_field    883 non-null    object
 7   skills           883 non-null    object
 8   job_description  883 non-null    object
 9   applicants       883 non-null    object
dtypes: object(10)
memory usage: 75.9+ KB


Также отметим, что в некоторых столбцах пропуски заполнены, например not specified	в столбцах с количеством работников и отраслью

Проверим столбцы с названиями стран и городов

In [27]:
data['location'].unique()

array(['Basel', 'Coventry', 'South Molton', 'Lugano', 'Southampton',
       'Leeds', 'Nuneaton', 'Paris', 'Cambridge', 'West Midlands',
       'Schiphol', 'Chester', 'Craven Arms', 'Dublin', 'Belfast',
       'Sunderland', 'Montévrain', 'Bristol', 'Solihull', 'Blackpool',
       'Cracow', 'Dijon', 'Alsónémedi', 'Manchester',
       'Elliniko-Argyroupoli', 'Italy', 'Umeå', 'North Holland',
       'Vilnius', 'Durham', 'Oudenaarde', 'Milan', 'Stockholm County',
       'Luxembourg', 'Roubaix', 'Munich', 'Zaventem',
       'Brussels Metropolitan Area', 'Portugal', 'West Malling',
       'Greater Paris Metropolitan Region', 'France', 'Bulgaria', 'Lille',
       'Egham', 'Karlstad', 'Madrid', 'Barcelona', 'Oxford', 'Hungary',
       'Brussels', 'Taibon', 'Epsom', 'Amsterdam', 'Spinea',
       'Greater Palma de Mallorca Metropolitan Area', 'Brindisi',
       'Boulogne-Billancourt', 'Wolfsburg', 'Nantes', 'Derby', 'Lund',
       'Garwolin', 'Stockholm', 'Rome', 'Massy', 'Prague',
       'Middle

Приведем названия локаций к единому виду

In [28]:
data['location']=data['location'].str.replace('Metropolitan','')
data['location']=data['location'].str.replace('Greater','')
data['location']=data['location'].str.replace('Area','')
data['location']=data['location'].str.replace('Region','')
data['location']=data['location'].str.replace('County','')

In [29]:
data['location']=data['location'].str.strip()

In [30]:
data['location'].unique()

array(['Basel', 'Coventry', 'South Molton', 'Lugano', 'Southampton',
       'Leeds', 'Nuneaton', 'Paris', 'Cambridge', 'West Midlands',
       'Schiphol', 'Chester', 'Craven Arms', 'Dublin', 'Belfast',
       'Sunderland', 'Montévrain', 'Bristol', 'Solihull', 'Blackpool',
       'Cracow', 'Dijon', 'Alsónémedi', 'Manchester',
       'Elliniko-Argyroupoli', 'Italy', 'Umeå', 'North Holland',
       'Vilnius', 'Durham', 'Oudenaarde', 'Milan', 'Stockholm',
       'Luxembourg', 'Roubaix', 'Munich', 'Zaventem', 'Brussels',
       'Portugal', 'West Malling', 'France', 'Bulgaria', 'Lille', 'Egham',
       'Karlstad', 'Madrid', 'Barcelona', 'Oxford', 'Hungary', 'Taibon',
       'Epsom', 'Amsterdam', 'Spinea', 'Palma de Mallorca', 'Brindisi',
       'Boulogne-Billancourt', 'Wolfsburg', 'Nantes', 'Derby', 'Lund',
       'Garwolin', 'Rome', 'Massy', 'Prague', 'Middlesbrough',
       'Viladecans', 'Warsaw', 'Eindhoven', 'Budapest', 'London',
       'Wroclaw', 'Amersfoort', 'Hamburg', 'Poland', 'The 

То же сделаем с названиями стран

In [31]:
data['country'].unique()

array(['Switzerland', 'United Kingdom', 'France', 'Netherlands',
       'Ireland', 'Poland', 'Hungary', 'Greece', 'Italy', 'Sweden',
       'Lithuania', 'Belgium', 'Luxembourg', 'Germany',
       'Brussels Metropolitan Area', 'Portugal',
       'Greater Paris Metropolitan Region', 'Bulgaria', 'Spain',
       'Greater Palma de Mallorca Metropolitan Area', 'Czechia',
       'Warsaw Metropolitan Area', 'Wroclaw Metropolitan Area', 'Latvia',
       'Estonia', 'Greater Barcelona Metropolitan Area', 'Malta',
       'Austria', 'Iasi Metropolitan Area', 'Cologne Bonn Region',
       'Greater Munster Area', 'Greater Banska Bystrica Area',
       'Greater Nuremberg Metropolitan Area', 'Norway',
       'Berlin Metropolitan Area', 'Denmark', 'Finland', 'Romania',
       'Rotterdam and The Hague', 'Prague Metropolitan Area',
       'Greater Pau Area', 'Monaco', 'Lisbon Metropolitan Area',
       'Athens Metropolitan Area', 'Greater Lyon Area',
       'Greater Milan Metropolitan Area', 'Croatia', 'S

In [32]:
data['country']=data['country'].str.replace('Metropolitan','')
data['country']=data['country'].str.replace('Greater','')
data['country']=data['country'].str.replace('Area','')
data['country']=data['country'].str.replace('Region','')

In [33]:
data['country']=data['country'].str.strip()

In [34]:
data['country'].unique()

array(['Switzerland', 'United Kingdom', 'France', 'Netherlands',
       'Ireland', 'Poland', 'Hungary', 'Greece', 'Italy', 'Sweden',
       'Lithuania', 'Belgium', 'Luxembourg', 'Germany', 'Brussels',
       'Portugal', 'Paris', 'Bulgaria', 'Spain', 'Palma de Mallorca',
       'Czechia', 'Warsaw', 'Wroclaw', 'Latvia', 'Estonia', 'Barcelona',
       'Malta', 'Austria', 'Iasi', 'Cologne Bonn', 'Munster',
       'Banska Bystrica', 'Nuremberg', 'Norway', 'Berlin', 'Denmark',
       'Finland', 'Romania', 'Rotterdam and The Hague', 'Prague', 'Pau',
       'Monaco', 'Lisbon', 'Athens', 'Lyon', 'Milan', 'Croatia',
       'Stuttgart', 'Slovakia', 'Dijon', 'Amsterdam', 'Oslo', 'Eindhoven',
       'Munich', 'Madrid', 'Krakow', 'Verona', 'Copenhagen'], dtype=object)

In [35]:
def new(row):
    country=row['country']
    if country in ['Brussels']: 
        return 'Belgium'
    elif country in ['Palma de Mallorca']: 
        return 'Spain'
    elif country in ['Warsaw', 'Wroclaw','Krakow']: 
        return 'Poland'
    elif country in ['Barcelona', 'Madrid']: 
        return 'Spain'
    elif country in ['Iasi']: 
        return 'Romania'
    elif country in ['Cologne Bonn', 'Munster','Nuremberg','Berlin','Stuttgart','Munich']: 
        return 'Germany'
    elif country in ['Banska Bystrica']: 
        return 'Slovakia'
    elif country in ['Rotterdam and The Hague', 'Amsterdam','Eindhoven']: 
        return 'Netherlands'
    elif country in ['Prague']: 
        return 'Czechia'
    elif country in ['Pau','Lyon', 'Dijon']: 
        return 'France'
    elif country in ['Lisbon']: 
        return 'Portugal'
    elif country in ['Athens']: 
        return 'Greece'
    elif country in ['Milan','Verona']: 
        return 'Italy'
    elif country in ['Oslo']: 
        return 'Norway'
    elif country in ['Copenhagen']: 
        return 'Denmark'
    else: return country


In [36]:
data['country']=data.apply(new, axis = 1)
data['country'].value_counts()

Italy             156
Germany           116
France            114
Netherlands        72
United Kingdom     70
Spain              63
Poland             59
Belgium            43
Portugal           26
Sweden             22
Ireland            16
Greece             16
Luxembourg         14
Bulgaria           14
Hungary            11
Austria             9
Czechia             8
Romania             8
Denmark             6
Latvia              5
Malta               5
Lithuania           5
Norway              4
Switzerland         4
Paris               4
Slovakia            4
Finland             3
Croatia             3
Estonia             2
Monaco              1
Name: country, dtype: int64

In [37]:
data['country']

0         Switzerland
2      United Kingdom
3      United Kingdom
4         Switzerland
5      United Kingdom
            ...      
993             Spain
994             Italy
995           Germany
996             Italy
997             Spain
Name: country, Length: 883, dtype: object

## Работа со столбцом  job_description

уберем лишние пробелы

In [38]:
data['job_description']=data['job_description'].str.strip()

In [39]:
data.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   
5         Data Analyst - Logistics   Southampton  United Kingdom   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   
5         On-site                      Butler, Bridge & May    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   
5            not specified   None   

                                     job_description     applicants  
0  What You Will Achieve   This position will app...           47.0  
2  Data Analyst - Logistics ~ Permanent Role ~ Mo...  not specified  
3  Salary:     To be discussed on application    ...  not specified  
4  FORFIRM     is providing solutions to real bus...  not specified  
5  Location: Southampton, HampshireJob title: Pla...  not specified

Приведем значения в столбце с названием вакансии к нижнему регистру

In [40]:
data['title']=data['title'].str.lower()

In [41]:
data['title'].str.lower().unique()

array(['data analyst', 'data analyst - logistics',
       'data analyst (space & planning)', 'data analyst - hybrid working',
       'commercial data analyst', 'asset data analyst',
       'data analyst - hybrid', 'data analyst (ft)', 'data analyst h/f',
       'data analyst ii', 'data analyst (9 months ftc)',
       'business intelligence analyst', 'data analyst (f/h)',
       'bi analyst', 'data analyst associate', 'product data analyst',
       'graduate data analyst', 'digital data analyst',
       'business intelligence analyst junior', 'data analyst (m/f/d)',
       'analytics consultant', 'data analyst f/h', 'consultor bi',
       'statistical data analyst', 'digital data analyst (h/f)',
       'data analyst - €60,- per hour - amsterdam based',
       'data & analytics senior analyst',
       'junior data analyst – aerospace', 'data analyst - operations',
       'data analyst (h/f)', 'business intelligence analyst (m/w/d)',
       'analista superior de datos', 'junior data analy

Отберем лишь релевантные заданию вакансии

In [42]:
data=data[data['title'].str.contains('data analyst|data scientist|datový analytik|bi analyst|business intelligence|data-analist|analityk danych|data analist|bi-analyst|analytics analyst|data analysis|data science analyst|analyste des donnees|data governance analyst|datenanalyst|analista de datos|analyste de données|data science|datový analytik/vývojář|data analytics|data & analytics|analyst data|ingeniero/a analista de datos big data|data specialist|data and analytics|analyste data|analityk|power bi')]

In [43]:
data.shape[0]

556

In [44]:
data.head()

title      location         country  \
0                     data analyst         Basel     Switzerland   
2         data analyst - logistics      Coventry  United Kingdom   
3  data analyst (space & planning)  South Molton  United Kingdom   
4                     data analyst        Lugano     Switzerland   
5         data analyst - logistics   Southampton  United Kingdom   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   
5         On-site                      Butler, Bridge & May    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   
5            not specified   None   

                                     job_description     applicants  
0  What You Will Achieve   This position will app...           47.0  
2  Data Analyst - Logistics ~ Permanent Role ~ Mo...  not specified  
3  Salary:     To be discussed on application    ...  not specified  
4  FORFIRM     is providing solutions to real bus...  not specified  
5  Location: Southampton, HampshireJob title: Pla...  not specified

Создадим столбец с требуемыми навыками, для этого приведем символы в описании вакансии к нижнему регистру

In [45]:
data['job_description']=data['job_description'].str.lower()

Выпишем названия программ

In [46]:
fg=[ 'a/b testing', 'ab testing', 'actian', 'adobe analytics', 'adobe audience manager',
    'adobe experience platform', 'adobe launch', 'adobe target', 'ai', 'airflow',
    'alooma', 'alteryx', 'amazon machine learning', 'amazon web services', 'aml',
    'amplitude', 'ansible', 'apache camel', 'apache nifi', 'apache spark',
    'api', 'asana', 'auth0', 'aws', 'aws glue', 'azure', 'azure data factory',
    'basecamp', 'bash', 'beats', 'big query', 'bigquery', 'birst', 'bitbucket',
    'blendo', 'bootstrap', 'business objects bi', 'c#', 'c++', 'caffe', 'cassandra',
    'cdata sync', 'chronograf', 'ci/cd', 'cicd', 'clickhouse', 'cloudera', 'cluvio',
    'cntk', 'cognos', 'composer', 'computer vision', 'conda', 'confluence',
    'couchbase', 'css', 'd3.js', 'dash', 'dashboard', 'data factory', 'data fusion',
    'data mining', 'data studio', 'data warehouse', 'databricks', 'dataddo',
    'dataflow', 'datahub', 'dataiku', 'datastage', 'dbconvert', 'dbeaver', 'dbt',
    'deep learning', 'dl/ml', 'docker', 'domo', 'dune', 'dv360', 'dynamodb',
    'elasticsearch', 'elt', 'erwin', 'etl', 'etleap', 'excel', 'facebook business manager',
    'fivetran', 'fuzzy', 'ga360', 'gcp', 'gensim', 'ggplot', 'git', 'github', 'gitlab',
    'google ads', 'google analytics', 'google cloud platform', 'google data flow',
    'google optimize', 'google sheets', 'google tag manager', 'google workspace',
    'grafana', 'hadoop', 'hana', 'hanagrafana', 'hbase', 'hdfs', 'hevo data', 'hightouch',
    'hive', 'hivedatabricks', 'html', 'hubspot', 'ibm coremetrics', 'inetsoft',
    'influxdb', 'informatica', 'integrate.io', 'iri voracity', 'izenda', 'java',
    'java script', 'javascript', 'jenkins', 'jira', 'jmp', 'julia', 'jupyter',
    'k2view', 'kafka', 'kantar', 'kapacitor', 'keras', 'kibana', 'kubernetes',
    'lambda', 'linux', 'logstash', 'looker', 'lstm', 'luidgi', 'matillion', 'matlab',
    'matplotlib', 'mendix', 'metabase', 'microsoft sql', 'microsoft sql server',
    'microstrategy', 'miro', 'mixpanel', 'ml', 'ml flow', 'mlflow', 'mongodb', 'mxnet',
    'mysql', 'natural nanguage processing', 'neo4j', 'nlp', 'nltk', 'nosql', 'numpy',
    'oauth', 'octave', 'omniture', 'omnituregitlab', 'openshift', 'openstack',
    'optimizely', 'oracle', 'oracle business intelligence', 'oracle data integrator',
    'pandas', 'panorama', 'pentaho', 'plotly', 'postgre', 'postgresql', 'posthog',
    'power amc', 'power bi', 'power point', 'powerbi', 'powerpivot', 'powerpoint',
    'powerquery', 'pyspark', 'python', 'pytorch', 'pytorchhevo data', 'qlik',
    'qlik sense', 'qlikview', 'querysurge', 'r', 'raphtory', 'rapidminer', 'redash',
    'redis', 'redshift', 'retool', 'rivery', 'rust', 's3', 'sa360', 'salesforce', 'sap',
    'sap business objects', 'sas', 'sas visual analytics', 'scala', 'scikit-learn',
    'scipy', 'seaborn', 'segment', 'selenium', 'sem rush', 'semrush', 'shell', 'shiny',
    'singer', 'sisense', 'skyvia', 'snowflake', 'spacy', 'spark', 'sparkml', 'splunk',
    'spotfire', 'spreadsheet', 'spss', 'sql', 'ssis', 'sssr', 'stambia', 'statistics',
    'statsbot', 'stitch', 'streamlit', 'streamsets', 'svn', 't-sql', 'tableau', 'talend',
    'targit', 'tealium', 'telegraf', 'tensorflow', 'terraapi', 'terraform', 'theano',
    'thoughtspot', 'timeseries', 'trello', 'unix', 'vba', 'vtom', 'webfocus', 'wfh',
    'xplenty', 'xtract.io', 'yellowfin'
          ]

Перед тем как выделять навыки подготовим данные, для этого уберем все запятые и скобки

In [47]:
data['job1']=data['job_description'].str.replace(',','')
data['job1']=data['job1'].str.replace(')','')
data['job1']=data['job1'].str.replace('(','')



Используем функцию split для разделения слов

In [48]:
data['job1']=data['job1'].str.split(' ')

In [49]:
data=data.reset_index()

что получилось в итоге:

In [50]:
data.head()

index                            title      location         country  \
0      0                     data analyst         Basel     Switzerland   
1      2         data analyst - logistics      Coventry  United Kingdom   
2      3  data analyst (space & planning)  South Molton  United Kingdom   
3      4                     data analyst        Lugano     Switzerland   
4      5         data analyst - logistics   Southampton  United Kingdom   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                       Mole Valley Farmers    not specified   
3         On-site                                   FORFIRM    not specified   
4         On-site                      Butler, Bridge & May    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description     applicants  \
0  what you will achieve   this position will app...           47.0   
1  data analyst - logistics ~ permanent role ~ mo...  not specified   
2  salary:     to be discussed on application    ...  not specified   
3  forfirm     is providing solutions to real bus...  not specified   
4  location: southampton, hampshirejob title: pla...  not specified   

                                                job1  
0  [what, you, will, achieve, , , this, position,...  
1  [data, analyst, -, logistics, ~, permanent, ro...  
2  [salary:, , , , , to, be, discussed, on, appli...  
3  [forfirm, , , , , is, providing, solutions, to...  
4  [location:, southampton, hampshirejob, title:,...

Напишем функцию которая будет сравнивать 2 списка и отбирать совпадения

In [51]:
def new1(row):
    job_description=row['job1']
    c = list(set(job_description) & set(fg))
    return c

In [52]:
data['job2']=data.apply(new1, axis = 1)

In [53]:
data.head()

index                            title      location         country  \
0      0                     data analyst         Basel     Switzerland   
1      2         data analyst - logistics      Coventry  United Kingdom   
2      3  data analyst (space & planning)  South Molton  United Kingdom   
3      4                     data analyst        Lugano     Switzerland   
4      5         data analyst - logistics   Southampton  United Kingdom   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                       Mole Valley Farmers    not specified   
3         On-site                                   FORFIRM    not specified   
4         On-site                      Butler, Bridge & May    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description     applicants  \
0  what you will achieve   this position will app...           47.0   
1  data analyst - logistics ~ permanent role ~ mo...  not specified   
2  salary:     to be discussed on application    ...  not specified   
3  forfirm     is providing solutions to real bus...  not specified   
4  location: southampton, hampshirejob title: pla...  not specified   

                                                job1  \
0  [what, you, will, achieve, , , this, position,...   
1  [data, analyst, -, logistics, ~, permanent, ro...   
2  [salary:, , , , , to, be, discussed, on, appli...   
3  [forfirm, , , , , is, providing, solutions, to...   
4  [location:, southampton, hampshirejob, title:,...   

                                       job2  
0  [statistics, sas, excel, spss, sql, sap]  
1                                     [wfh]  
2                                   [excel]  
3      [etl, python, oracle, gcp, aws, sql]  
4                                   [excel]

Оценим датасет, пропусков нет

In [54]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 556 entries, 0 to 555
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   index            556 non-null    int64 
 1   title            556 non-null    object
 2   location         556 non-null    object
 3   country          556 non-null    object
 4   employment_type  556 non-null    object
 5   company_name     556 non-null    object
 6   employee_qty     556 non-null    object
 7   company_field    556 non-null    object
 8   skills           556 non-null    object
 9   job_description  556 non-null    object
 10  applicants       556 non-null    object
 11  job1             556 non-null    object
 12  job2             556 non-null    object
dtypes: int64(1), object(12)
memory usage: 56.6+ KB


Подготовим данные для создания дашборда

In [55]:
data=data.explode('job2').reset_index(drop=True)

In [56]:
data.head()

index         title location      country employment_type  \
0      0  data analyst    Basel  Switzerland         On-site   
1      0  data analyst    Basel  Switzerland         On-site   
2      0  data analyst    Basel  Switzerland         On-site   
3      0  data analyst    Basel  Switzerland         On-site   
4      0  data analyst    Basel  Switzerland         On-site   

                               company_name     employee_qty  \
0  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
2  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
3  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
4  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   

             company_field skills  \
0  Staffing and Recruiting   None   
1  Staffing and Recruiting   None   
2  Staffing and Recruiting   None   
3  Staffing and Recruiting   None   
4  Staffing and Recruiting   None   

                                     job_description applicants  \
0  what you will achieve   this position will app...       47.0   
1  what you will achieve   this position will app...       47.0   
2  what you will achieve   this position will app...       47.0   
3  what you will achieve   this position will app...       47.0   
4  what you will achieve   this position will app...       47.0   

                                                job1        job2  
0  [what, you, will, achieve, , , this, position,...  statistics  
1  [what, you, will, achieve, , , this, position,...         sas  
2  [what, you, will, achieve, , , this, position,...       excel  
3  [what, you, will, achieve, , , this, position,...        spss  
4  [what, you, will, achieve, , , this, position,...         sql

Оценим количество уникальных значений в столбце job2

In [57]:
data['job2'].value_counts()

sql         252
python      200
excel       169
tableau     116
ai          110
           ... 
lambda        1
fivetran      1
splunk        1
grafana       1
t-sql         1
Name: job2, Length: 127, dtype: int64

Всего 127 шт, некоторые из них встречааются всего один раз, есть смысл такие удалить

In [58]:
ks=data.groupby('job2')['job1'].count().loc[lambda x : x > 3].reset_index().drop('job1', axis=1)
ks=ks['job2'].to_list()

In [59]:
data=data.query('job2 in @ks')

In [60]:
data['job2'].unique()

array(['statistics', 'sas', 'excel', 'spss', 'sql', 'sap', 'etl',
       'python', 'oracle', 'gcp', 'aws', 'tableau', 'nosql', 'qlikview',
       'powerbi', 'scala', 'spotfire', 'spreadsheet', 'redshift', 'api',
       'r', 'talend', 'informatica', 'pandas', 'bigquery', 'jupyter',
       'looker', 'dbt', 'ai', 'salesforce', 'powerpoint', 'dashboard',
       'spark', 'cognos', 'jira', 'airflow', 'ci/cd', 'hive', 'hadoop',
       'kafka', 'alteryx', 'elt', 'snowflake', 'metabase',
       'microstrategy', 'qlik', 'numpy', 'azure', 'javascript', 'java',
       'git', 'mysql', 'databricks', 'confluence', 'vba', 'c#',
       'scikit-learn', 'ssis', 'matlab', 'powerquery', 'powerpivot',
       'julia', 'docker', 'nlp', 'pytorch', 'tensorflow', 'c++',
       'postgresql', 'ml'], dtype=object)

In [61]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1862 entries, 0 to 2007
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   index            1862 non-null   int64 
 1   title            1862 non-null   object
 2   location         1862 non-null   object
 3   country          1862 non-null   object
 4   employment_type  1862 non-null   object
 5   company_name     1862 non-null   object
 6   employee_qty     1862 non-null   object
 7   company_field    1862 non-null   object
 8   skills           1862 non-null   object
 9   job_description  1862 non-null   object
 10  applicants       1862 non-null   object
 11  job1             1862 non-null   object
 12  job2             1862 non-null   object
dtypes: int64(1), object(12)
memory usage: 203.7+ KB


In [62]:
data=data.drop(['skills','job_description','job1'], axis=1)

In [63]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1862 entries, 0 to 2007
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   index            1862 non-null   int64 
 1   title            1862 non-null   object
 2   location         1862 non-null   object
 3   country          1862 non-null   object
 4   employment_type  1862 non-null   object
 5   company_name     1862 non-null   object
 6   employee_qty     1862 non-null   object
 7   company_field    1862 non-null   object
 8   applicants       1862 non-null   object
 9   job2             1862 non-null   object
dtypes: int64(1), object(9)
memory usage: 160.0+ KB


In [64]:
data['job2'].unique()

array(['statistics', 'sas', 'excel', 'spss', 'sql', 'sap', 'etl',
       'python', 'oracle', 'gcp', 'aws', 'tableau', 'nosql', 'qlikview',
       'powerbi', 'scala', 'spotfire', 'spreadsheet', 'redshift', 'api',
       'r', 'talend', 'informatica', 'pandas', 'bigquery', 'jupyter',
       'looker', 'dbt', 'ai', 'salesforce', 'powerpoint', 'dashboard',
       'spark', 'cognos', 'jira', 'airflow', 'ci/cd', 'hive', 'hadoop',
       'kafka', 'alteryx', 'elt', 'snowflake', 'metabase',
       'microstrategy', 'qlik', 'numpy', 'azure', 'javascript', 'java',
       'git', 'mysql', 'databricks', 'confluence', 'vba', 'c#',
       'scikit-learn', 'ssis', 'matlab', 'powerquery', 'powerpivot',
       'julia', 'docker', 'nlp', 'pytorch', 'tensorflow', 'c++',
       'postgresql', 'ml'], dtype=object)

In [65]:
def new2(row):
    job2=row['job2']
    if job2 in ['pandas','numpy','scikit-learn']:
        return 'python'
    elif job2 in ['vba','powerquery','powerpivot']:
        return 'excel'
    elif job2 in ['postgresql','mysql']:
        return 'sql'
    else: 
        return job2

In [66]:
data['job2']=data.apply(new2, axis = 1)

In [67]:
data['job2'].unique()

array(['statistics', 'sas', 'excel', 'spss', 'sql', 'sap', 'etl',
       'python', 'oracle', 'gcp', 'aws', 'tableau', 'nosql', 'qlikview',
       'powerbi', 'scala', 'spotfire', 'spreadsheet', 'redshift', 'api',
       'r', 'talend', 'informatica', 'bigquery', 'jupyter', 'looker',
       'dbt', 'ai', 'salesforce', 'powerpoint', 'dashboard', 'spark',
       'cognos', 'jira', 'airflow', 'ci/cd', 'hive', 'hadoop', 'kafka',
       'alteryx', 'elt', 'snowflake', 'metabase', 'microstrategy', 'qlik',
       'azure', 'javascript', 'java', 'git', 'databricks', 'confluence',
       'c#', 'ssis', 'matlab', 'julia', 'docker', 'nlp', 'pytorch',
       'tensorflow', 'c++', 'ml'], dtype=object)

In [68]:
data.pivot_table(index='employment_type',values='title', aggfunc=lambda x: len(x.unique()))

title
employment_type       
Hybrid             131
On-site             96
Remote              28
not specified       25

In [69]:
data.to_csv('data.csv')

## Код для сопоставления и проверки данных 

In [70]:
data[(data['country']=='Greece') & (data['location']=='Athens')]

index                          title location country employment_type  \
951     290  data scientist - data analyst   Athens  Greece         On-site   
952     290  data scientist - data analyst   Athens  Greece         On-site   
953     290  data scientist - data analyst   Athens  Greece         On-site   
954     290  data scientist - data analyst   Athens  Greece         On-site   
955     290  data scientist - data analyst   Athens  Greece         On-site   
956     290  data scientist - data analyst   Athens  Greece         On-site   
957     290  data scientist - data analyst   Athens  Greece         On-site   
958     290  data scientist - data analyst   Athens  Greece         On-site   
959     290  data scientist - data analyst   Athens  Greece         On-site   
961     290  data scientist - data analyst   Athens  Greece         On-site   
963     290  data scientist - data analyst   Athens  Greece         On-site   
964     290  data scientist - data analyst   Athens  Greece         On-site   
1018    354            online data analyst   Athens  Greece          Remote   
1239    518                 data scientist   Athens  Greece          Hybrid   
1240    518                 data scientist   Athens  Greece          Hybrid   
1241    518                 data scientist   Athens  Greece          Hybrid   
1363    577        data analytics engineer   Athens  Greece          Hybrid   
1364    577        data analytics engineer   Athens  Greece          Hybrid   
1365    577        data analytics engineer   Athens  Greece          Hybrid   
1366    577        data analytics engineer   Athens  Greece          Hybrid   
1367    577        data analytics engineer   Athens  Greece          Hybrid   
1479    624                data specialist   Athens  Greece          Hybrid   
1827    865                 data scientist   Athens  Greece          Hybrid   
1828    865                 data scientist   Athens  Greece          Hybrid   

               company_name           employee_qty  \
951                      EY      10,001+ employees   
952                      EY      10,001+ employees   
953                      EY      10,001+ employees   
954                      EY      10,001+ employees   
955                      EY      10,001+ employees   
956                      EY      10,001+ employees   
957                      EY      10,001+ employees   
958                      EY      10,001+ employees   
959                      EY      10,001+ employees   
961                      EY      10,001+ employees   
963                      EY      10,001+ employees   
964                      EY      10,001+ employees   
1018    TELUS International      10,001+ employees   
1239             Alpha Bank      10,001+ employees   
1240             Alpha Bank      10,001+ employees   
1241             Alpha Bank      10,001+ employees   
1363                    GWI    501-1,000 employees   
1364                    GWI    501-1,000 employees   
1365                    GWI    501-1,000 employees   
1366                    GWI    501-1,000 employees   
1367                    GWI    501-1,000 employees   
1479  Modelity Technologies       51-200 employees   
1827                  efood  1,001-5,000 employees   
1828                  efood  1,001-5,000 employees   

                             company_field     applicants        job2  
951                  Professional Services  not specified        spss  
952                  Professional Services  not specified     tableau  
953                  Professional Services  not specified    spotfire  
954                  Professional Services  not specified        java  
955                  Professional Services  not specified      python  
956                  Professional Services  not specified    qlikview  
957                  Professional Services  not specified      hadoop  
958                  Professional Services  not specified      oracle  
959                  Profession

In [71]:
data[data['country']=='Greece']

index                           title              location country  \
68       30          data analyst associate  Elliniko-Argyroupoli  Greece   
69       30          data analyst associate  Elliniko-Argyroupoli  Greece   
70       30          data analyst associate  Elliniko-Argyroupoli  Greece   
71       30          data analyst associate  Elliniko-Argyroupoli  Greece   
72       30          data analyst associate  Elliniko-Argyroupoli  Greece   
73       30          data analyst associate  Elliniko-Argyroupoli  Greece   
74       30          data analyst associate  Elliniko-Argyroupoli  Greece   
951     290   data scientist - data analyst                Athens  Greece   
952     290   data scientist - data analyst                Athens  Greece   
953     290   data scientist - data analyst                Athens  Greece   
954     290   data scientist - data analyst                Athens  Greece   
955     290   data scientist - data analyst                Athens  Greece   
956     290   data scientist - data analyst                Athens  Greece   
957     290   data scientist - data analyst                Athens  Greece   
958     290   data scientist - data analyst                Athens  Greece   
959     290   data scientist - data analyst                Athens  Greece   
961     290   data scientist - data analyst                Athens  Greece   
963     290   data scientist - data analyst                Athens  Greece   
964     290   data scientist - data analyst                Athens  Greece   
1018    354             online data analyst                Athens  Greece   
1061    403             online data analyst              Iráklion  Greece   
1176    465             online data analyst          Thessaloniki  Greece   
1205    501             online data analyst              Rethymno  Greece   
1239    518                  data scientist                Athens  Greece   
1240    518                  data scientist                Athens  Greece   
1241    518                  data scientist                Athens  Greece   
1363    577         data analytics engineer                Athens  Greece   
1364    577         data analytics engineer                Athens  Greece   
1365    577         data analytics engineer                Athens  Greece   
1366    577         data analytics engineer                Athens  Greece   
1367    577         data analytics engineer                Athens  Greece   
1479    624                 data specialist                Athens  Greece   
1518    648  business intelligence engineer       Palaión Fáliron  Greece   
1519    648  business intelligence engineer       Palaión Fáliron  Greece   
1520    648  business intelligence engineer       Palaión Fáliron  Greece   
1521    648  business intelligence engineer       Palaión Fáliron  Greece   
1549    669             online data analyst                Sparta  Greece   
1827    865                  data scientist                Athens  Greece   
1828    865                  data scientist                Athens  Greece   

     employment_type                              company_name  \
68           On-site                    LAMDA Development S.A.   
69           On-site                    LAMDA Development S.A.   
70           On-site                    LAMDA Development S.A.   
71           On-site                    LAMDA Development S.A.   
72           On-site                    LAMDA Development S.A.   
73           On-site                    LAMDA Development S.A.   
74           On-site                    LAMDA Development S.A.   
951          On-site                                        EY   
952          On-site                                        EY   
953          On-site                                        EY   
954          On-site                                        EY   
955          On-site                                        EY   
956          On-site                                        EY   
9

In [72]:
data['index'].nunique()

502

In [73]:
data[data['company_name']=='Revolut']

index                                title         location    country  \
325      92              data analyst (engineer)           Madrid      Spain   
326      92              data analyst (engineer)           Madrid      Spain   
327      92              data analyst (engineer)           Madrid      Spain   
328      92              data analyst (engineer)           Madrid      Spain   
329      92              data analyst (engineer)           Madrid      Spain   
330      92              data analyst (engineer)           Madrid      Spain   
605     173               data analyst - finance           Madrid      Spain   
606     173               data analyst - finance           Madrid      Spain   
607     173               data analyst - finance           Madrid      Spain   
608     173               data analyst - finance           Madrid      Spain   
609     173               data analyst - finance           Madrid      Spain   
1185    468  data analyst (product data analyst)           Poland     Poland   
1186    468  data analyst (product data analyst)           Poland     Poland   
1187    468  data analyst (product data analyst)           Poland     Poland   
1372    581               data analyst - finance           Poland     Poland   
1373    581               data analyst - finance           Poland     Poland   
1374    581               data analyst - finance           Poland     Poland   
1375    581               data analyst - finance           Poland     Poland   
1376    581               data analyst - finance           Poland     Poland   
1491    634              data analyst (engineer)           Poland     Poland   
1492    634              data analyst (engineer)           Poland     Poland   
1493    634              data analyst (engineer)           Poland     Poland   
1494    634              data analyst (engineer)           Poland     Poland   
1495    634              data analyst (engineer)           Poland     Poland   
1496    634              data analyst (engineer)           Poland     Poland   
1686    775  data analyst (product data analyst)           Cracow     Poland   
1687    775  data analyst (product data analyst)           Cracow     Poland   
1688    775  data analyst (product data analyst)           Cracow     Poland   
1723    789               data analyst - finance           Cracow     Poland   
1724    789               data analyst - finance           Cracow     Poland   
1725    789               data analyst - finance           Cracow     Poland   
1726    789               data analyst - finance           Cracow     Poland   
1727    789               data analyst - finance           Cracow     Poland   
1754    817              data scientist (credit)           Poland     Poland   
1755    817              data scientist (credit)           Poland     Poland   
1756    817              data scientist (credit)           Poland     Poland   
1772    835              data scientist (credit)        Lithuania  Lithuania   
1773    835              data scientist (credit)        Lithuania  Lithuania   
1774    835              data scientist (credit)        Lithuania  Lithuania   
1818    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1819    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1820    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1821    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1822    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1823    860      credit - data analyst (product)         Bulgaria   Bulgaria   
1907    914      credit - data analyst (product)            Porto   Portugal   
1908    914      credit - data analyst (product)            Porto   Portugal   
1909    914      credit - data analyst (product)            Porto   Portugal   
1910    914      credit - data analyst (product)            Porto   Portugal   
1911  

## Дашборд

Первая часть работы, а точнее предобработка данных подошла к концу, можно приступать к созданию дашборда. Дашборд можно увидеть по ссылке: https://public.tableau.com/app/profile/steve2592/viz/Master_16864027105980/sheet8?publish=yes